compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -51,18 +47,21 @@ object TypeApplications {
         // In this case, we can still return true if we know that the hk lambda bounds
         // are empty anyway.
       || {
+        val tparams = fn.typeParams
         val paramRefs = tparams.map(_.paramRef)
+        val prefix = fn match { case fn: TypeRef => fn.prefix       case _ => NoPrefix }
+        val owner  = fn match { case fn: TypeRef => fn.symbol.owner case _ => NoSymbol }
         tp.typeParams.corresponds(tparams) { (param1, param2) =>
-          param2.paramInfo frozen_<:< param1.paramInfo.substParams(tp, paramRefs)
+          param2.paramInfo.asSeenFrom(prefix, owner) frozen_<:< param1.paramInfo.substParams(tp, paramRefs)
```

Comment:
Should be able to use `paramInfoAsSeenFrom`.

Reply:
Doesn't use the correct owner.